src/conn.rs

```diff
@@ -287,8 +306,24 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
                 }
                 () = &mut idle_timeout => {
                     if !std::matches!(self.state, State::Closed { .. }) {
+                        // Warn that we are giving up on the connection due to a lack of activity.
+
+                        // First, we look for this special situation:
+                        // Our outgoing FIN was sent, but the incoming ACK was dropped.
+                        // This timeout is not worthy of a warning log, because everything
+                        // basically went fine: all data was sent and ACK'd.
+                        // There is no more work to do.
+
                         let unacked: Vec<u16> = self.unacked.keys().copied().collect();
-                        tracing::warn!(?unacked, "idle timeout expired, closing...");
+                        let finished = match self.state {
+                            State::Closing { local_fin, remote_fin, .. } => unacked.len() == 1 && local_fin.is_some() && &local_fin.unwrap() == unacked.last().unwrap() && remote_fin.is_some(),
+                            _ => false,
+                        };
+                        if finished {
+                            tracing::debug!(?self.state, ?unacked, "Idle timeout with both FINs sent, with only FIN unacked.");
```

Comment:
prefer lowercase and no terminal punctuation

```suggestion
                            tracing::debug!(?self.state, ?unacked, "idle timeout expired, but only missing ACK for local FIN");
```